language/stdlib/modules/Offer.move

```diff
@@ -35,6 +35,93 @@ module Offer {
   public fun address_of<Offered>(offer_address: address): address acquires T {
     borrow_global<T<Offered>>(offer_address).for
   }
+
+  /**
+   **************** SPECIFICATIONS ****************
+
+  This module defines a resource T that is used as a permissioned trading scheme between accounts.
+  It defines two main functions for creating and retrieving a struct offered by some user
+  inside the resource T under the offerer's account.
+
+  Currently, the only other module that depends on this module is LibraConfig, where it's used to
+  pass a capability to an account that allows it to modify a config.
+  */
+
+  /// # Module specification
+
+  spec module {
+    /// Verify all functions in this module
+    pragma verify = true;
+    /// Helper function that returns whether or not the `recipient` is an intended
+    /// recipient of the offered struct in the T<Offered> resource at the address `offer_address`
+    define is_offer_recipient<Offered>(offer_addr: address, recipient: address): bool {
+      recipient == global<T<Offered>>(offer_addr).for || recipient == offer_addr
+    }
+  }
+
+  /// ## Creation of Offers
+
+  spec schema OnlyCreateCanCreateOffer {
+    /// Only `Self::create` can create an Offer.T under an address.
+    ///
+    /// **Informally:** No function can create an offer. If there didn't
```

Comment:
The explanation is a bit misleading because it's not "no function" but "no function to which this is applied..."

My explanations when I've used this pattern need to be improved, also.